src/gc-alloc-profiler.cpp

```diff
@@ -49,15 +49,22 @@ jl_combined_results g_combined_results; // Will live forever.
 // === stack stuff ===
 
 jl_raw_backtrace_t get_raw_backtrace() JL_NOTSAFEPOINT {
-    // A single large buffer to record backtraces onto
-    static jl_bt_element_t static_bt_data[JL_MAX_BT_SIZE];
+    // We first record the backtrace onto a MAX-sized buffer, so that we don't have to
+    // allocate the buffer until we know the size. To ensure thread-safety, we use a
+    // per-thread backtrace buffer.
+    jl_ptls_t ptls = jl_current_task->ptls;
+    jl_bt_element_t *shared_bt_data_buffer = ptls->profiling_bt_buffer;
```

Comment:
```suggestion
    jl_bt_element_t *shared_bt_data_buffer = ptls->profiling_bt_buffer;
    if (shared_bt_data_buffer == NULL)
        ptls->profiling_bt_buffer = shared_bt_data_buffer = \
                malloc_s(sizeof(jl_bt_element_t) * (JL_MAX_BT_SIZE + 1));
```

Reply:
oops - yes, thank you! We discussed doing exactly this yesterday, but i forgot to get to it. 🙏